modules/javafx.controls/src/test/java/test/javafx/scene/control/ListCellStartEditTest.java

```diff
@@ -0,0 +1,126 @@
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.scene.control;
+
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.control.cell.CheckBoxListCell;
+import javafx.scene.control.cell.ChoiceBoxListCell;
+import javafx.scene.control.cell.ComboBoxListCell;
+import javafx.scene.control.cell.TextFieldListCell;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Collection;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * Parameterized tests for the {@link ListCell#startEdit()} method of {@link ListCell} and all sub implementations.
+ * The {@link CheckBoxListCell} is special as in there the checkbox will be disabled
+ * based of the editability.
+ */
+@RunWith(Parameterized.class)
+public class ListCellStartEditTest {
+
+    private static final boolean[] EDITABLE_STATES = { true, false };
+
+    private ListView<String> listView;
+    private final ListCell<String> listCell;
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        return wrapAsObjectArray(List.of(new ListCell<>(), new ComboBoxListCell<>(), new TextFieldListCell<>(),
+                new ChoiceBoxListCell<>(), new CheckBoxListCell<>(obj -> new SimpleBooleanProperty())));
+    }
+
+    private static Collection<Object[]> wrapAsObjectArray(List<ListCell<?>> listCells) {
+        return listCells.stream().map(tc -> new Object[] { tc }).collect(toList());
+    }
+
+    public ListCellStartEditTest(ListCell<String> listCell) {
+        this.listCell = listCell;
+    }
+
+    @Before
+    public void setup() {
+        ObservableList<String> items = FXCollections.observableArrayList("1", "2", "3");
+        listView = new ListView<>(items);
+    }
+
+    @Test
+    public void testStartEdit() {
+        // First test startEdit() without anything set yet.
+        listCell.startEdit();
+
+        listCell.updateIndex(0);
+
```

Comment:
for my taste, this method is doing too much: the first line is the real subject of the issue (throwing an NPE), the rest of the method is testing the other issue (make sure that editability constraints are respected for all combinations). 

My suggestion: separate them out into different methods, f.i. 

- testStartEditMustNotThrowNPE (containing only the single offending line)
- testStartEditRespectsEditable (containing the rest)
- not yet covered: sanity test that startEdit really installs the editing visuals - f.i. by checking that its graphic is null before/ not-null after startEdit